### PR TITLE
Create/delete posts

### PR DESCRIPTION
**Pull request by VangSang0**
Creating and deleting posts:

- app.py has some functionalities
- some html as well on the homepage
- created a few new html's as well

Please check every one of those out

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -121,7 +104,14 @@ def logout():
 @app.get('/home-page')
 @other_methods.check_user # This is a decorator that checks if the user is signed in
 def home_page():
-    return render_template('home_page.html', posts=posts)
+    all_posts = database_methods.get_all_posts()
+    random.shuffle(all_posts)
+    for post in all_posts:
```
Comment (linalu12) [inline app.py:109]: randomizing posts and user in session look good!

File: app.py [hunk 2]
```diff
@@ -138,6 +128,68 @@ def friends():
...
 
+
+
+@app.post('/user-post')
+@other_methods.check_user
+def user_post():
+    post_author_id = database_methods.get_user_by_id(session['user_id'])['user_id']
+    post_content = request.form.get('post-content')
+    post_id = database_methods.add_post(post_author_id, post_content)
+    if post_id is None:
```
Comment (linalu12) [inline app.py:139]: perhaps add  " and if post-content is None"  in test case

File: app.py [hunk 3]
```diff
@@ -138,6 +128,68 @@ def friends():
...
+
+@app.get('/posts/<int:post_id>')
+@other_methods.check_user
+def post(post_id):
+    post = database_methods.get_post_by_id(post_id)
+    if post is None:
+        abort(404)
+    post['datetime_post'] = other_methods.format_datetime(post['datetime_post'])
+    return render_template('posts.html', post=post)
+
```
Comment (linalu12) [inline app.py:152]: lgtm!

File: app.py [hunk 4]
```diff
@@ -138,6 +128,68 @@ def friends():
...
+    
+    print(post_id)
+    print(post)
+    if post is None:
+        flash('There is no post with that ID')
+    if post['post_author_id'] != session['user_id']:
+        flash("You are not authorized to delete this post")
+        return redirect(url_for('home_page'))
+    
+    database_methods.delete_post(post_id)
```
Comment (linalu12) [inline app.py:168]: lgtm

File: app.py [hunk 5]
```diff
@@ -138,6 +128,68 @@ def friends():
...
+@other_methods.check_user
+def user_posts():
+    user_id = session['user_id']
+    user = database_methods.get_user_by_id(user_id)
+    print(user)
+    user_posts = database_methods.get_posts_by_user_id(user_id)
+    for post in user_posts:
+        post['datetime_post'] = other_methods.format_datetime(post['datetime_post'])
+    return render_template('users_posts.html', user=user, user_posts=user_posts)
+
```
Comment (linalu12) [inline app.py:192]: everything looks great Sang!

File: repositories/database_methods.py [hunk 6]
```diff
@@ -68,3 +68,130 @@ def get_user_by_username(username: str) -> dict[str, Any] | None:
...
+    pool = get_pool()
+    with pool.connection() as connection:
+        with connection.cursor() as cursor:
+            cursor.execute('''
+                            DELETE FROM 
+                                Posts
+                            WHERE
+                                post_id = %s
+                            ''', [post_id])
+            # what can I return here?
```
Comment (linalu12) [inline repositories/database_methods.py:197]: queries lgtm

File: static/script.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (linalu12) [inline static/script.js:1]: nice event listeners for post edit

**Review by linalu12 — CHANGES_REQUESTED** [review]
only left one possible change under the add post method, I think you should add something regarding if the content of the post is empty

**Comment by dali-t** [comment]
lgtm!

**Pull request merged**